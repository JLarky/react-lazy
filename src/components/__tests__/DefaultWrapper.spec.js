/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'

import DefaultWrapper from '../DefaultWrapper'

test('DefaultWrapper className indicates placeholder state by default', () => {
    const tree = renderer.create(<DefaultWrapper />)
    expect(tree.toJSON()).toMatchSnapshot()
})

test('DefaultWrapper className indicates loading state when childProps exists', () => {
    const tree = renderer.create(<DefaultWrapper childProps={{}} />)
    expect(tree.toJSON()).toMatchSnapshot()
})

test('DefaultWrapper className indicates failed state when isFailed', () => {
    const tree = renderer.create(<DefaultWrapper childProps={{}} isFailed />)
    expect(tree.toJSON()).toMatchSnapshot()
})

test('DefaultWrapper className indicates loaded state when isLoaded', () => {
    const tree = renderer.create(<DefaultWrapper childProps={{}} isLoaded />)
    expect(tree.toJSON()).toMatchSnapshot()
})

test('DefaultWrapper may contain children', () => {
    const tree = renderer.create(
        <DefaultWrapper childProps={{}} isLoaded>
            <div />
        </DefaultWrapper>,
    )
    expect(tree.toJSON()).toMatchSnapshot()
})

test('DefaultWrapper passes other props directly to the rendered div element', () => {
    const tree = renderer.create(<DefaultWrapper childProps={{}} isLoaded id="you-should-see-me-in-the-div" />)
    expect(tree.toJSON()).toMatchSnapshot()
})
