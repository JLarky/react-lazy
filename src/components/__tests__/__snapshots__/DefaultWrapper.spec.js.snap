// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DefaultWrapper className indicates failed state when isFailed 1`] = `
<div
  className="react-lazy-wrapper react-lazy-wrapper--failed"
/>
`;

exports[`DefaultWrapper className indicates loaded state when isLoaded 1`] = `
<div
  className="react-lazy-wrapper react-lazy-wrapper--loaded"
/>
`;

exports[`DefaultWrapper className indicates loading state when childProps exists 1`] = `
<div
  className="react-lazy-wrapper react-lazy-wrapper--loading"
/>
`;

exports[`DefaultWrapper className indicates placeholder state by default 1`] = `
<div
  className="react-lazy-wrapper react-lazy-wrapper--placeholder"
/>
`;

exports[`DefaultWrapper may contain children 1`] = `
<div
  className="react-lazy-wrapper react-lazy-wrapper--loaded"
>
  <div />
</div>
`;

exports[`DefaultWrapper passes other props directly to the rendered div element 1`] = `
<div
  className="react-lazy-wrapper react-lazy-wrapper--loaded"
  id="you-should-see-me-in-the-div"
/>
`;
